### ISSUE TITLE
Default the DirectWrite setting to be ON

### ISSUE THREAD

**Issue by alankilborn** [notepad-plus-plus/notepad-plus-plus#13198]
### Description of the Issue
A case has been made to change the default setting for the _Use DirectWrite_ to be checkmarked.  Users complain time and time again that their unicode characters aren't being displayed correctly, and have to be pointed to the direct-write setting.  Please lessen the tech-support burden and default this feature to ON:

![image](https://user-images.githubusercontent.com/77065706/220725618-d7d0a947-8065-4da3-bc37-5e94307d27a2.png)

### Steps to Reproduce the Issue
1. Enter a certain set of unicode character(s) -- for a reasonable reproduction, see this Community thread and seek out the  `Feb 21, 2023, 1:54 PM` posting: https://community.notepad-plus-plus.org/topic/24081/notepad-v8-4-9-release

### Expected Behavior
Expect to see characters rendered correctly, without having to change any settings.

### Actual Behavior
1. Characters rendered incorrectly.
2. User complains on support site.
3. User is directed to the _DirectWrite_ setting.
4. User is dissatisfied that they had to waste their time figuring out this problem.
5. People providing tech support get frustrated (how many times do we have to answer this question?)

### Debug Information
Notepad++ v8.4.9   (64-bit)
Build time : Jan 27 2023 - 03:11:16
Path : W:\npp.misc\_NoBackup\Releases\npp.8.4.9\npp.8.4.9.portable.x64\notepad++.exe
Command Line : -titleAdd="8.4.9 64-bit" -multiInst
Admin mode : OFF
Local Conf mode : ON
Cloud Config : OFF
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 21H2
OS Build : 19044.2486
Current ANSI codepage : 1252
Plugins : 
    Explorer (1.9.3)
    mimeTools (2.9)
    NppConverter (4.5)
    NppExport (0.4)
    PythonScript (3.0.14)

**Comment by Yaron10** [comment]
I just want to point out that DirectWrite breaks RTL (View -> Text Direction LTR/RTL).

**Comment by rdipardo** [comment]
> I just want to point out that DirectWrite breaks RTL (View -> Text Direction LTR/RTL).

:+1:

It also breaks *every* Scintilla 5 API that takes a `colouralpha` LPARAM, e.g. [`SCI_SETELEMENTCOLOUR`] and friends: https://github.com/d0vgan/nppexec/issues/69#issuecomment-1165491420

The latter issue is really the fault of N++'s retrograde preference for 24-bit [COLORREF]s, but don't expect to see a migration to [GDI+] and alpha colours anytime soon, judging from the reception I got [the last time I suggested it].

[`SCI_SETELEMENTCOLOUR`]: https://www.scintilla.org/ScintillaDoc.html#SCI_SETELEMENTCOLOUR
[COLORREF]:https://learn.microsoft.com/en-us/windows/win32/gdi/colorref
[GDI+]: https://learn.microsoft.com/en-us/windows/win32/gdiplus/-gdiplus-gdi-start
[the last time I suggested it]: https://github.com/notepad-plus-plus/notepad-plus-plus/pull/11744#discussion_r890610856

**Comment by Yaron10** [comment]
@rdipardo,

Thank you for the additional info. 👍

**Comment by Devocub** [comment]
It also breaks fonts? But I actually prefer "broken" version of font (it looks normal).  
Substantial effect - Notepad++ startup tim with DirectWrite enabled 500 ms vs 270 ms with disabled. Is it really that bad? What it doing these additional 230 ms?

**Comment by alankilborn** [comment]
>Notepad++ startup tim with DirectWrite enabled 500 ms vs 270 ms with disabled. Is it really that bad? What it doing these additional 230 ms?

This data needs to be quantified.
How are you measuring this?
Do you have files that are in the current session on startup? The size of these files could affect such timings.

**Comment by Devocub** [comment]
@alankilborn just recorded screen (starting notepad exe from cmd when cmd blinking cursor is in ON state - it disappears when you press enter), repeat 3 times. Even without measurement this additional delay is clearly visible

**Comment by Devocub** [comment]
How to check if font is broken: I tried [iosevka](https://github.com/be5invis/Iosevka/releases/download/v24.1.3/super-ttc-iosevka-24.1.3.zip) all fonts.  
With DirectWrite ON they look mostly same when switching over different variants.  
With DirectWrite write OFF they all look different.  
So I wanted to find what is the actual font notepad++ is using intead of broken one (i tried iosevka extended, which looks different (=correct) with directwrite off). Turns out it was...  Leelawade UI! Few fonts after iosevka. Why? Feels like maybe notepad skip fonts going down until it finds working one, though fonts between iosevka and leelawade work? Something weird  is happening

**Comment by rdipardo** [comment]
> i tried iosevka extended, which looks different (=correct) with directwrite off

Known issue: https://sourceforge.net/p/scintilla/bugs/2356

**Comment by donho** [comment]
https://github.com/notepad-plus-plus/notepad-plus-plus/commit/975d29b30bfd624b6392541b133c2bd32f3212b0#commitcomment-132946815

**Comment by Devocub** [comment]
@donho sad to see this decision. I think adding note near DirectWrite setting that it might slowdown startup twice (250 → 500 ms for me) is useful.

**Comment by Devocub** [comment]
@donho 
Assuming Windows, with animations turned off (just search it in settings) (but even with animations turned on difference is clear).
1. You have cursor blinking in cmd every second, when you press enter it disappears and typed command executed.
2. Start recording screen with OBS (60 or more fps, preferably using hardware encoding like NVENC or AMD/Intel HW encoders so cpu load will affect the experiment less)
3. Type Notepad++.exe (or its full path) in cmd. When blinking cursor in state ON press Enter. The blinking cursor will disappear early. Be careful not to be slow.
4. After Notepad++ appears, stop OBS recording and using some video editor measure time between when you pressed Enter (when blinking cursor disappears early in cmd) and when you see something from Notepad++, preferably text.
Try with DirectWrite ON and OFF. Difference should be clear and reproducible.

If you don't have video editor maybe use online ones, or download (for example: https://github.com/mltframework/shotcut , didn't try this one).
If you fail to catch cursor in state ON I can just send you my videos.
If you don't observe difference - share the videos to make sure you measured properly.
If you can't measure time in video editor, cut out precisely everything before Enter and after Notepad++, do this for both variants (DirectWrite ON and OFF) and compare their lengths visually.
You can try reducing CPU frequency using power settings in Windows but it shouldn't be necessary.

I hope that was detailed enough, if you still fail - ditch it. I have no problem for me while I can have DirectWrite OFF.

**Comment by Devocub** [comment]
@xomx yeah, it shows correctly, +200 ms to Notepad++ initialization with DirectWrite ON. 140 ms → 340 ms.
  
Note, I edited measurement guide: 0. Set power plan to high performance or lock cpu freq.

**Comment by xomx** [comment]
@Devocub 
> 340 ms

I see similar values, but independent on the DirectWrite setting (below are calculated averages from two consequent measurements, used fresh N++ v8.6.2 x64 portable with default settings):

Win10 Ent LTSC 21H2 x64 (desktop PC)
DirectWrite ON ~303ms
DirectWrite OFF ~311ms

Win11 Pro 22H2 x64 (laptop)
DirectWrite ON ~334ms
DirectWrite OFF ~339ms

Maybe you have some sort of caching mechanism behind (which I do not have) and the DirectWrite could interfere with that.

**Comment by Devocub** [comment]
@xomx It's related to some windows bloatware in your system. I just tried 21H2 (on VM) and had same results as you. Disabling windows defender (all hidden settings too) and font cache service didn't make any changes (towards faster) so then I applied my big debloat scripts which disable a lot of stuff in windows and then notepad++ init time went down to values expected from OS running on VM. I don't know exactly what solved it but If you want to investigate try searching "debloat windows github".

**Comment by Devocub** [comment]
I tested this again on freshest win 10 (22H2 19045.4412)
DW OFF 135 ms, DW ON 209 ms. these are fastest results. Now this speed is acceptable. Still recommend making DW OFF as default because DW ON produces incorrect fonts which is more imortant rather than properly displayed emojis.
  
I also tested Leelawade UI vs Iosevka Extended fonts. What was bothering me - Leelawade UI was slow when I had a lot of symbols displayed, I had noticeable lag when typing and this led to mistyping. Turns out Iosevka Extended is fast in these cases so would like to see fonts speed benchmark.
